File: app-api/API/DatabaseLayer/Models/Area.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DatabaseLayer.Models
{
    public class Area
    {
        public int AreaId { get; set; }
        public string Name { get; set; }

        public int CityId { get; set; }
        public virtual City City { get; set; }

        public virtual ICollection<Product> Products { get; set; }
    }
}

File: Mobile App (Flutter)/trade_it/lib/models/area.dart
import 'package:trade_it/models/product.dart';

class Area{
  late int id;
  late String name;
  List<Product>? products;

  Area({
    required this.id,
    required this.name,
    this.products
  });

  factory Area.fromJson(Map<String,dynamic> json){
    Area area = Area(
      id: json['areaId'],
      name: json['name'],
      products: json['product'] == null ? null : List<Product>.from(json['products'].map((product)=>Product.fromJson(product))),
    );

    return area;
  }
}